src/parser/mod.rs

```diff
@@ -11384,6 +11406,23 @@ impl<'a> Parser<'a> {
         })
     }
 
+    /// Parses MSSQL's json-null-clause
+    fn parse_json_null_clause(&mut self) -> Option<JsonNullClause> {
+        if !dialect_of!(self is MsSqlDialect) {
+            return None;
+        }
```

Comment:
here we can use a dialect method as well?

Reply:
Maybe we can remove this predication?

Reply:
Yeah I think we can do that, since the syntax is unambiguous enough